ampd/src/handlers/multisig.rs

```diff
@@ -278,6 +275,44 @@ mod test {
         .unwrap()
     }
 
+    // this returns an event that is named SigningStarted,
+    // but the fields are different than what the handler expects
+    fn wrong_signing_started_event() -> events::Event {
+        let pub_keys = (0..10)
+            .map(|_| (rand_account().to_string(), rand_public_key()))
+            .collect::<HashMap<String, PublicKey>>();
+
+        let poll_started = SigningStarted {
+            session_id: Uint64::one(),
+            worker_set_id: "worker_set_id".to_string(),
+            pub_keys,
+            msg: MsgToSign::unchecked(rand_message()),
+            chain_name: rand_chain_name(),
+            expires_at: 100u64,
+        };
+
+        let mut event: cosmwasm_std::Event = poll_started.into();
+        event.ty = format!("wasm-{}", event.ty);
+        event = event.add_attribute("_contract_address", MULTISIG_ADDRESS);
+        let idx = event
+            .attributes
+            .iter()
+            .position(|element| element.key == "expires_at")
+            .unwrap();
+        event.attributes.remove(idx);
```

Comment:
- there must be a simpler way to remove a specific element
- this doesn't look like an event with wrong fields, but an event with a missing field, or do I see that wrong?

Reply:
yes it's just a missing field. I still consider this an event with wrong fields, but it can be renamed. This is the actual scenario I was running into: I deployed a new version of the contract that added a field, and events emitted from the old contract were crashing the handler.

Reply:
yeah I'd prefer a more specific name